compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -394,6 +394,21 @@ impl<'a> Parser<'a> {
             } else {
                 PatKind::Path(qself, path)
             }
+        } else if matches!(self.token.kind, token::Lifetime(_))
+            && !self.look_ahead(1, |token| matches!(token.kind, token::Colon))
```

Comment:
In pattern position, we're totally fine with using "next token isn't colon" as a heuristic. We could probably just _always_ try to recover if it's a lifetime, because we never have `'a: label {}` in a pattern position anyways.

Reply:
Could we include this as a comment?

Reply:
Can do